lib/partition_setup.pm

```diff
@@ -327,6 +327,7 @@ sub addboot {
 }
 
 sub select_first_hard_disk {
+    return send_key $cmd{next} if (check_screen('select-hard-disks-one-selected', 0));
     my $matched_needle = assert_screen [qw(existing-partitions hard-disk-dev-sdb-selected hard-disk-dev-non-sda-selected)];
```

Comment:
Just add this needle to this assert_screen call and return if `match_has_tag`

Reply:
I miss that, thanks, but anyways we need to click next also.

Reply:
We already have early return in the method, if you will add a comment I would prefer to have early return than another level of nesting, as for me personally, that is harder to read. What if we extract that code to separate function?

Reply:
you're right, this two returns in the same function would not smell good neither, but in another subroutine should be better.